cfg_mgmt/util.py

```diff
@@ -297,7 +301,31 @@ def rotate_config_element_and_persist_in_cfg_repo(
         git_helper.add_and_commit(
             message=commit_message,
         )
-        git_helper.push('@', target_ref)
+        for attempt in range(1, MAX_RETRIES + 1):
```

Comment:
I suggest to stick to using recursion (as we commonly do in other places in our codebase). like so:

```
def function_with_retry(retries=123):
  try:
   something()
  except ExpectedException as e:
   if retries > 0:
     function_with_retry(retries=retries - 1)
```